DOC: update the DataFrame.cov docstring

sechilds opened the pull request:
Update the docstring with some examples from
elsewhere in the pandas documentation.

Some of the examples use randomly generated time series
because we need to get covariance between long series.
Used a random seed to ensure that the results are the
same each time.

Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.cov" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

sechilds committed:
DOC: Revise docstring of DataFrame cov method

Update the docstring with some examples from
elsewhere in the pandas documentation.

Some of the examples use randomly generated time series
because we need to get covariance between long series.
Used a random seed to ensure that the results are the
same each time.

TomAugspurger commented on pandas/core/frame.py:
```diff
@@ -5511,12 +5526,71 @@ def cov(self, min_periods=None):
+        See Also
+        --------
+        Series.cov : compute covariance with another Series
+        core.window.EWM.cov: expoential weighted sample covariance
```
These should all start with `pandas.` so that the are linked correctly.

TomAugspurger commented on pandas/core/frame.py:
```diff
@@ -5511,12 +5526,71 @@ def cov(self, min_periods=None):
+        **Minimum number of periods**
+
+        This method also supports an optional ``min_periods`` keyword
+        that specifies the required minimum number of observations for each
```
maybe non-NA observations, just to be 100% clear :)

TomAugspurger reviewed (commented):
Minor comment about the See Also. LGTM otherwise.

sechilds committed:
DOC: Fix See Also and min_periods explanation.

Responding to comments on PR. See also section will link
properly and number of periods explanation clearer.

TomAugspurger commented:
This looks very good. Thanks!
